torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2431,54 +2431,79 @@ Tensor linalg_qr_backward(const std::vector<torch::autograd::Variable> &grads, c
   }
 }
 
-// Invertible case is derived from Jacobi's formula, and also can be found at:
-// http://eprints.maths.ox.ac.uk/1079/1/NA-08-01.pdf
 Tensor linalg_det_backward(const Tensor & grad, const Tensor& self, const Tensor& det) {
-  auto singular_case_backward = [&](const Tensor& grad, const Tensor& self, const Tensor& det) -> Tensor {
-    Tensor u, sigma, v;
-    std::tie(u, sigma, v) = self.svd();
-    auto gsigma = prod_backward(grad.unsqueeze(-1), sigma, det.unsqueeze(-1));
-    return svd_backward({{}, gsigma, {}}, self, true, true, u, sigma, v);
+  auto det_backward_nonsingular = [&](const Tensor& grad, const Tensor& self, const Tensor& det) -> Tensor {
+    // Derived from Jacobi's formula for partial derivative, which can be
+    // found at https://en.wikipedia.org/wiki/Jacobi%27s_formula
+    auto a = self.transpose(-2, -1);
+    auto b = unsqueeze_multiple(det, {-1, -2}, self.dim()) * at::eye(a.size(-1), det.options());
```

Comment:
I feel like there might be a more performant way to do this. I thought `torch.diag()` would do it, but it doesn't seem so

Reply:
I found out I can restride and then use `diag_embed`

Reply:
What is written here essentially is `det * I`, so we can first make `det` a Tensor of a single scalar, then `expand` it, then perform a `diag_embed` of it.

Reply:
Yep, `diag_embed` works on non-contiguous tensors. The `multiple_unsqueeze` and `expand` can be replaced by a single `restride_as` call